### PR TITLE
Feature/us2 tasks

### LINKED ISSUE #86: Only use active nodes to determine the network version
image

Should be 100% on `0.12.0` because all `active` nodes are on that version.

### PR DESCRIPTION
**Pull request by mbartn**
* removed `local_storage_service` for now because it was buggy, i made a mistake to initialize it with the context which should be dynamic for each call, let's use `user_data dict` inline for now
* refresh nodes in `user_data` from network on every restart
* fix #86 
* refactoring

### CODE REVIEW

File: bot/handlers/network_info.py [hunk 1]
```diff
@@ -23,21 +23,16 @@ async def show_network_stats(update, context):
...
-        versions = {}
-        for validator in validators:
-            status = validator['status']
-            version = validator['version']
-
-            statuses[status] = 1 if not status in statuses else statuses[status] + 1
-            versions[version] = 1 if not version in versions else versions[version] + 1
+        statuses_counter = Counter(map(lambda v: v['status'], validators))
+        active_validators = filter(lambda v: v['status'] == 'active', validators)
+        versions_counter = Counter(map(lambda v: v['version'], active_validators))
```
Comment (mr-rooftop) [inline bot/handlers/network_info.py:28]: SCOUT Bartnicki, you left the place behind clean af!

File: bot/thornode_bot.py [hunk 2]
```diff
@@ -103,6 +90,32 @@ def setup_existing_user(dispatcher):
...
+    # Delete all node addresses and add them again to ensure all user_data fields are up to date
+    for chat_id in dispatcher.user_data.keys():
+        if 'nodes' not in dispatcher.user_data[chat_id]:
+            dispatcher.user_data[chat_id]['nodes'] = {}
+
+        local_node_addresses = list(dispatcher.user_data[chat_id]['nodes'].keys())
+
+        for address in local_node_addresses:
+            try:
+                new_node = next(n for n in new_node_accounts if n['address'] == address)
```
Comment (mr-rooftop) [inline bot/thornode_bot.py:108]: On this line I get a key error
```
KeyError: 'address'
```
I think we need "node_address" instead

**Review by mr-rooftop — CHANGES_REQUESTED** [review]

File: bot/thornode_bot.py (hunk 2 above)
Comment (mbartn) [reply bot/thornode_bot.py]: Good catch, no idea why i did it this way

File: bot/thornode_bot.py [hunk 3]
```diff
@@ -103,6 +90,32 @@ def setup_existing_user(dispatcher):
             if os.path.exists(session_data_path):
                 os.remove(session_data_path)
 
+    try:
+        new_node_accounts = get_node_accounts()
+    except:
+        logger.exception("Fatal error! I couldn't get node accounts to update the local user_data!", exc_info=True)
+        return
```
Comment (mr-rooftop) [inline bot/thornode_bot.py:97]: When I run the testcases, here is always the excpetion thrown:
```
constants - ERROR - Fatal error! I couldn't get node accounts to update the local user_data!
.
.
ConnectionRefusedError: [Errno 111] Connection refused
```

Seems like the mock_api is not ready yet. 
Don't know why this happens only with the integration tests, and not when I run the bot normally.
I added `time.sleep(3)` after the debug process instantiation which solved the error throwing.

**Review by mr-rooftop — APPROVED** [review]

File: bot/thornode_bot.py (hunk 3 above)
Comment (mbartn) [reply bot/thornode_bot.py:75]: I decided to go with `sleep()`, it's not really elegant however we want to abandon the mock api process anyway in the future

**Pull request merged**